src/lib/input/input.ts

```diff
@@ -311,7 +311,14 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
 
   /** Focuses the input. */
   focus(): void {
+    // fix IE11 not able to focus programmatically with css style -ms-user-select: none
+    // see https://github.com/angular/material2/issues/15093
+    this._elementRef.nativeElement.style[<any>'-ms-user-select'] = 'auto';
+
     this._elementRef.nativeElement.focus();
+
+    // reset IE11 fix css style
+    this._elementRef.nativeElement.style[<any>'-ms-user-select'] = '';
```

Comment:
Rather than doing this in JS, it might be better if we either reset the `-ms-user-select` in CSS or we never emitted it to begin with. This styling is there as a nice-to-have anyway.

Reply:
Yeah I think we should just remove the problematic style. Doing this will allow people to focus it by calling `.focus()` on the component, but it could still prevent focus in other situations

Reply:
@crisbeto @mmalerba thanks for the review. I updated the PR according to your suggestions to override the problematic styling in CSS for IE only. Could you please have another look at this PR?